template_config.yml

```diff
@@ -1,33 +1,36 @@
 # This config represents the latest values used when running the plugin-template. Any settings that
 # were not present before running plugin-template have been added with their default values.
 
-# generated with plugin_template@2021.04.08-87-g2136629-dirty
+# generated with plugin_template@2021.04.08-96-g0697947
 
 additional_repos:
-  - name: pulp_ansible
-    org: pulp
-    bindings: true
-    branch: 0.9.0
-  - name: pulp_container
-    org: pulp
-    bindings: true
-    branch: 2.7.1
-  - name: galaxy-importer
-    bindings: false
-    org: ansible
-    branch: v0.3.4
+- bindings: true
+  revision: "0b6440a9ddd1048fd1d6a8945c969a7c28d2a4c9"
+  branch: main
+  name: pulp_ansible
+  org: pulp
```

Comment:
@rochacbruno if we push a PR with a `galaxy_ng` change that is compatible with `pulp_ansible` master at this revision, it will pass CI. But if that `galaxy_ng` change conflicts with the `pulp_ansible` version that is pinned in `setup.py`, do we have any way of knowing that?

Reply:
@awcrosby no, this PR makes master to be always the bleeding edge of repos, the versions pinned on setup.py for those 4 repos are bypassed and requirements files are completely ignored.

You have a good point and what we can do is:

```bash
if PR sent against master branch (${{ github.event.pull_request.base.ref == 'master' }}) 
then
   install from latest compatible main branches
else
   install from pinned versions from setup.py and requirements files
fi
```

with that logic on `.github/workflows/scripts/post_before_install.sh` we can ensure that a PR sent against other branches `stable-x.y.z` will always use pinned versions, but master will always be the latest.

We need master to catch latest updates to keep us in sync with pulp, but we cannot break this compatibility by adding a code change that is compatible only with previous versions. We will only know when the PR is sent against a stable-... branch.

Reply:
The other option, would be to have another branch called `next` and this PR would go there, and changes targetting `next` pulp release would be done there, this will generate the need to backport/cherry pick multiple things.

Reply:
this also would require more customization to pulp/plugin_template @fao89 @daviddavis I think we can add a conditional version matrix to the `pulpcore_branch|revision` and `additional_repos.branch|revision`.

e.g:

```yaml
pulpcore_branch:
    - name: master
      when: "${{ github.event.pull_request.base.ref == 'master' }}"
    - name: 3.14
      when: "${{ github.event.pull_request.base.ref != 'master' }}"

additional_repos:
    - name: pulp_ansible
      branch:
        - name: main
          when: "${{ github.event.pull_request.base.ref == 'master' }}"
        - name: 0.7.0
          when: "${{ github.event.pull_request.base.ref != 'master' }}"
```

then we need to add a new job to each item on matrix and condition goes to the job after it is rendered by Jinja

Reply:
We just talked about it and decided that we are adding this as a nightly build and not as a job to run for every PR, I will work on the changes.